### DISCUSSION TITLE
Create Apple Note

### DISCUSSION THREAD

**Discussion by burhanrashid52** [johnlindquist/kit#1490] [Share]
Although we can do the same using Apple Shortcuts on Mac, I find it to be too slow and the window for typing notes is very small. This is my initial version; perhaps we can convert the Markdown version from the editor to Apple Notes formatting.

I already have a script for note creation and I would love to contribute to the Community Script if it's a common use case.

```ts
// Name: Create Apple Note
// Shortcut: command option control a

import "@johnlindquist/kit"

import * as applescript from 'applescript';

function createAppleNoteScript(content: string) {
  const title = 'Note created on ' + new Date().toLocaleDateString();
  return `
        tell application "Notes"
            tell account "iCloud"
                tell folder "Notes"
                    make new note with properties {name:"${title}", body:"${content}"}
                end tell
            end tell
        end tell
    `;
}

function createAppleNote(content: string) {
  const script = createAppleNoteScript(content);

  applescript.execString(script, (err: Error | null, result: any) => {
    if (err) {
      console.error('Error creating note:', err);
    } else {
      console.log('Note created successfully:', result);
    }
  });
}

const text = await editor("");
createAppleNote(text);
```

**Comment by aloshka** [comment]
Did you mean to attach a script?

**Reply by burhanrashid52** [reply]
I have a script that uses AppleScript to create Apple Notes. My question is, if this is a common use case, would it be helpful to share this script in the Community Script?

**Reply by aloshka** [reply]
Always. I'm sure there is someone that will find it VERY useful.

**Reply by burhanrashid52** [reply]
Okay Thanks. Now how can I contribute to Community Script ?

**Reply by burhanrashid52** [reply]
@aloshka I've updated the the comment with script above.
